src/payment/strategies/braintree/braintree-sdk-creator.ts

```diff
@@ -43,19 +51,54 @@ export default class BraintreeSDKCreator {
                 this.getClient(),
                 this._braintreeScriptLoader.loadPaypal(),
             ])
-            .then(([client, paypal]) => paypal.create({ client }));
+                .then(([client, paypal]) => paypal.create({ client }));
         }
 
         return this._paypal;
     }
 
-    getPaypalCheckout(): Promise<BraintreePaypalCheckout> {
+    getPaypalCheckout(
+        paypalOptions: BraintreePaypalButtonInitializeOptions,
+        paymentMethod: PaymentMethod,
+        offerCredit: boolean,
+        setupPayment: any,
+        tokenizePayment: any,
+        container: string):
+        Promise<BraintreePaypalCheckout> {
         if (!this._paypalCheckout) {
             this._paypalCheckout = Promise.all([
                 this.getClient(),
                 this._braintreeScriptLoader.loadPaypalCheckout(),
             ])
-                .then(([client, paypalCheckout]) => paypalCheckout.create({ client }));
+                .then(([client, paypalCheckout]) => paypalCheckout.create({ client }, (_error: string, instance: any) =>  {
+                     instance.loadPayPalSDK(() => {
+                             const allowedSources = [];
+                             const disallowedSources = [];
+
+                             if (paypalOptions.allowCredit) {
+                                 allowedSources.push(this._window && this._window.paypal && this._window.paypal.FUNDING.CREDIT);
```

Comment:
Per our conversation,
it will push bool, however, it should push funding source  

`(this._window && this._window.paypal && this._window.paypal.FUNDING.CREDIT)`

Reply:
fixed